xdsl/dialects/stim/ops.py

```diff
@@ -1,24 +1,119 @@
-from abc import ABC
+from collections.abc import Sequence
 from io import StringIO
 
-from xdsl.ir import Region
+from xdsl.dialects.builtin import ArrayAttr, IntAttr
+from xdsl.dialects.stim.stim_printer_parser import StimPrintable, StimPrinter
+from xdsl.ir import ParametrizedAttribute, Region, TypeAttribute
 from xdsl.irdl import (
     IRDLOperation,
-    PyRDLOpDefinitionError,
+    ParameterDef,
+    irdl_attr_definition,
     irdl_op_definition,
     region_def,
 )
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
 
-from .stim_printer_parser import StimPrintable, StimPrinter
 
+class StimAttr(StimPrintable):
+    "Base Stim attribute"
 
-class StimOp(IRDLOperation, ABC):
-    def print_stim(self, printer: StimPrinter) -> None:
-        raise (PyRDLOpDefinitionError("print_stim not implemented!"))
+    name = "stim.attr"
```

Comment:
What is the purpose of this class?

Reply:
Everything in this file is stim printable - in mlir the convention seems to be making a base class for the attributes and operations - is it okay to not do that here?

Reply:
We generally try to do it only when necessary. In this case, the fact that it has a name, and is a base class without inheriting from ABC are both a little confusing. I would just remove it for the time being.